src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelJsonProtocol.scala

```diff
@@ -325,4 +327,35 @@ object ModelJsonProtocol extends WorkspaceJsonSupport {
   implicit val impCreateRawlsBillingProjectFullRequestFormat = jsonFormat2(CreateRawlsBillingProjectFullRequest)
   implicit val impSpreadsheetResponse = jsonFormat1(SpreadsheetResponse)
 
+  implicit object ShareFormat extends RootJsonFormat[Share] {
+    override def write(obj: Share): JsValue = JsObject(Map(
+      "userId" -> JsString(obj.userId),
+      "sharee" -> JsString(obj.sharee),
+      "shareType" -> JsString(obj.shareType),
+      "timestamp" -> JsNumber(obj.timestamp.toEpochMilli) // not sure if this is the right conversion!
+    ))
+
+    override def read(json: JsValue): Share = json match {
+      case JsObject(jso) =>
+        val userId: String = jso.get("userId") match {
+          case Some(JsString(s)) => s
+          case _ => throw new DeserializationException("could not deserialize Share")
+        }
+        val sharee: String = jso.get("sharee") match {
+          case Some(JsString(s)) => s
+          case _ => throw new DeserializationException("could not deserialize Share")
+        }
+        val shareType: String = jso.get("shareType") match {
+          case Some(JsString(s)) => s
+          case _ => throw new DeserializationException("could not deserialize Share")
+        }
```

Comment:
I know I gave you this code :) There's a lot of cut-and-paste here, maybe extract a common method for these three?